lib/pure/uri.nim

```diff
@@ -8,6 +8,30 @@
 #
 
 ## This module implements URI parsing as specified by RFC 3986.
+##
+## A Uniform Resource Identifier (URI) provides a simple and extensible
+## means for identifying a resource. A URI can be further classified
+## as a locator, a name, or both. The term “Uniform Resource Locator”
+## (URL) refers to the subset of URIs.
+##
+## .. code-block::
+##    import uri
+##    let blog = "/blog.html"
+##    let host = parseUri("https://nim-lang.org")
+##    let bloguri = host / blog
+##    echo "Main page: " & $host
+##    # --> Main page: https://nim-lang.org
+##    echo "Blog page: " & $bloguri
+##    # --> Blog page: https://nim-lang.org/blog.html
```

Comment:
assert please

Reply:
Changed to `doAssert`. Is there standard for when to use `assert` and `doAssert` in code samples and runnableExamples?

Reply:
> Changed to doAssert. Is there standard for when to use assert and doAssert in code samples and runnableExamples?

Heh, there was a long discussion on IRC yesterday about that. 

There were no concrete conclusions, other than that when it comes to `runnableExamples`, `assert` is always checked when creating the documentation so there's no need to use `doAssert` (which feels "noisier" and more unfamiliar to Nim newcomers) — I switched to using `assert` in `runnableExamples`.

Reply:
Changes `doAssert` to `assert` in runnables and code samples.

f8e807229c225144544f5e02045553ddff991c23